### ISSUE TITLE
Read-Only support

### ISSUE THREAD

**Issue by abluejelly** [LeaVerou/awesomplete#16901]
Requesting the addition of a check to see if the input is marked as Read Only, and behaving appropriately if it is, as you can currently change a `readonly` field if its current value contained in multiple suggestions (obviously a bad thing).

Example:
list contains `aaa`,`aaaa`, and `aaaaa`
current value is `aaa`
field is marked `readonly`
If I click on the field, the popup will suggest all three options, and if I select any of them, it will change the chosen value.

I personally added it via slapping `if (this.input.readOnly) return;` at the top of evaluate(), as that allows it to be changed programmatically at practically any point (additional keyboard events included), but someone else might see/know a better spot for it.

**Comment by LeaVerou** [comment]
Yes, this is definitely a bug that needs to be fixed. Nice catch @abluejelly!
